src/io/mod.rs

```diff
@@ -92,11 +93,11 @@ impl Io {
 
     /// Opens a file for reading.  Transparently handles reading gzipped files based
     /// extension.
-    pub fn new_reader<P>(&self, p: &P) -> Result<Box<dyn BufRead + Send>>
+    pub fn new_reader<P>(&self, p: P) -> Result<Box<dyn BufRead + Send>>
```

Comment:
why take ownership?

Reply:
It's an `AsRef<Path>` - this means you can pass either an owned value or a reference. You are leaving it up to the caller to decide which they want to do. If you always want a reference, then you should just use `&Path`.